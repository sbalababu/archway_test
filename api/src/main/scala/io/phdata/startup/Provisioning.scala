/* Copyright 2018 phData Inc. */

package io.phdata.startup

import cats.effect._
import cats.implicits._
import io.phdata.config.ProvisioningConfig
import com.typesafe.scalalogging.LazyLogging
import io.phdata.AppContext
import io.phdata.services.ProvisioningService

class Provisioning[F[_]: Sync: Timer: ContextShift](context: AppContext[F], provisioningService: ProvisioningService[F])
    extends ScheduledJob[F] with LazyLogging {

  val provisioningConfig: ProvisioningConfig = context.appConfig.provisioning

  override def work: F[Unit] =
    for {
      _ <- logger.info("rectifying unprovisioned workspaces").pure[F]
      _ <- provisioningService.provisionAll()
      _ <- logger.info("provisioning going to sleep for {}", provisioningConfig.provisionInterval).pure[F]
      _ <- Timer[F].sleep(provisioningConfig.provisionInterval)
      _ <- work
    } yield ()

}
