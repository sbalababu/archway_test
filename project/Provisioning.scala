import sbt._
import sbt.Keys._

object Provisioning {
  
  val provisioningSettings =
    Seq(
      addCompilerPlugin("org.scalamacros" % "paradise" % "2.1.0" cross CrossVersion.full),
      libraryDependencies ++= Dependencies.provisioningDependencies
    )
  
}
