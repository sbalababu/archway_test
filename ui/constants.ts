export const TOKEN_EXTRACTOR = (s: any) => s.getIn(['login', 'token']);

export const RECENT_WORKSPACES_KEY = 'recentWorkspaces';
export const SPNEGO = 'spnego';
export const workspaceStatuses = ['approved', 'pending', 'rejected'];
export const workspaceBehaviors = ['simple', 'structured', 'custom'];
export const behaviorProperties = {
  simple: {
    title: 'Simple',
    icon: 'team',
  },
  structured: {
    title: 'Structured',
    icon: 'deployment-unit',
  },
  custom: {
    title: 'Custom',
    icon: 'select',
  },
};

export enum FeatureFlagType {
  Application = 'application',
  Messaging = 'messaging',
  DiskQuota = 'disk-quota',
}

export enum ProvisioningType {
  Pending = 'pending',
  Complete = 'complete',
}

export enum ModalType {
  DeprovisionWorkspace = 'deprovisionWorkspace',
  ProvisionWorkspace = 'provisionWorkspace',
  DeleteWorkspace = 'deleteWorkspace',
  Application = 'application',
  Kafka = 'kafka',
  SimpleTopicMember = 'simpleTopicMember',
  SimpleMember = 'simpleMember',
  ChangeOwner = 'changeOwner',
  ModifyDiskQuota = 'modifyDiskQuota',
  ModifyCoreMemory = 'modifyCoreMemory',
}

export enum NotificationType {
  Success = 'success',
  Info = 'info',
  Warning = 'warning',
  Error = 'error',
}

export enum ComplianceType {
  PCI = 'pci',
  PII = 'pii',
  PHI = 'phi',
}

export enum LinksGroupCardPage {
  Overview = 'home',
  Manage = 'manage',
}
