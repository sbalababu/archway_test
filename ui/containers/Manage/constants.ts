export enum ManagePage {
  ComplianceTab = 'compliances',
  LinksTab = 'custom-links',
}

export enum OpenType {
  Add = 'add',
  Update = 'update',
  Delete = 'delete',
}

export const formLayout = {
  labelCol: {
    xs: { span: 24 },
    sm: { span: 4 },
  },
  wrapperCol: {
    xs: { span: 24 },
    sm: { span: 20 },
    lg: { span: 16 },
  },
};

export const linkLayout = {
  labelCol: {
    xs: { span: 24 },
    sm: { span: 3 },
  },
  wrapperCol: {
    xs: { span: 24 },
    md: { span: 21 },
    lg: { span: 15 },
  },
};
