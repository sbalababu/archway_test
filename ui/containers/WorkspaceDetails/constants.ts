export enum ShowTypes {
  Users = 'Users',
  Groups = 'Groups',
}
