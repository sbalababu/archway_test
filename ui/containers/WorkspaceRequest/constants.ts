export const PAGE_BEHAVIOR = 'behavior';
export const PAGE_CUSTOM_WORKSPACES = 'custom workspaces';
export const PAGE_DETAILS = 'details';
export const PAGE_COMPLIANCE = 'compliance';
export const PAGE_REVIEW = 'review';
