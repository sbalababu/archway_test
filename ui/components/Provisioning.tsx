import * as React from 'react';
import { Icon } from 'antd';

import { ProvisioningType } from '../constants';

interface Props {
  provisioning: ProvisioningType;
}

const Provisioning = ({ provisioning }: Props) => (
  <div style={{ display: 'flex', alignItems: 'center', justifyContent: 'center', paddingTop: 8 }}>
    <Icon
      type={provisioning === ProvisioningType.Pending ? 'clock-circle' : 'check-circle'}
      theme="twoTone"
      style={{ fontSize: 20, paddingRight: 8 }}
    />
    <span style={{ fontSize: 14 }}>{`PROVISIONING: ${provisioning.toUpperCase()}`}</span>
  </div>
);

export default Provisioning;
