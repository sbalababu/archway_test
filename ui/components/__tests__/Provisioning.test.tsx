import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Provisioning from '../Provisioning';
import { ProvisioningType } from '../../constants';

describe('Provisioning', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Provisioning provisioning={ProvisioningType.Complete} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
