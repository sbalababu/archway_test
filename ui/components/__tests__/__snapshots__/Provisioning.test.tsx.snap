// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Provisioning renders correctly 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "justifyContent": "center",
      "paddingTop": 8,
    }
  }
>
  <Icon
    style={
      Object {
        "fontSize": 20,
        "paddingRight": 8,
      }
    }
    theme="twoTone"
    type="check-circle"
  />
  <span
    style={
      Object {
        "fontSize": 14,
      }
    }
  >
    PROVISIONING: COMPLETE
  </span>
</div>
`;
